src/main/resources/assets/computercraft/lua/rom/programs/edit.lua

```diff
@@ -14,8 +14,10 @@ if fs.exists( sPath ) and fs.isDir( sPath ) then
 end
 
 -- Create .lua files by default
-if not fs.exists( sPath ) and not string.find( sPath, "%." ) then
-	sPath = sPath .. ".lua"
+if settings.get( "edit.extension" ) then
+   if not fs.exists( sPath ) and not string.find( sPath, "%." ) then
```

Comment:
You're also using 3 spaces here and a tab character then 3 spaces on the next line. Seeing as the rest of the file uses 4 spaces, you should probably change this. 

This isn't the first time I've said this on your PRs, so can you please turn whitespace highlighting on in your editor - having a consistent whitespace makes it easier to read and edit the code in the future.